src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingService.kt

```diff
@@ -47,30 +53,63 @@ class PrisonPersonMoveBookingService(
 
   private suspend fun syncFromOffenderDps(fromOffenderNo: String, telemetry: MutableMap<String, String>) {
     physicalAttributesNomisApiService.getPhysicalAttributes(fromOffenderNo).also { nomisResponse ->
-      val latestBooking = nomisResponse.bookings.find { booking -> booking.latestBooking }
-      if (latestBooking != null) {
-        physicalAttributesSyncService.physicalAttributesChanged(fromOffenderNo, latestBooking.bookingId, nomisResponse)
-        telemetry += "syncFromOffenderDps_HEIGHT" to "true"
-        telemetry += "syncFromOffenderDps_WEIGHT" to "true"
-      }
+      nomisResponse.bookings
+        .find { it.latestBooking }
+        ?.also { latestBooking ->
+          physicalAttributesSyncService.physicalAttributesChanged(fromOffenderNo, latestBooking.bookingId, nomisResponse)
+          telemetry += "syncFromOffenderDps_HEIGHT" to "true"
+          telemetry += "syncFromOffenderDps_WEIGHT" to "true"
+        }
     }
 
-    // TODO SDIT-2200 sync all profile details for from offender
+    profileDetailsNomisApiService.getProfileDetails(fromOffenderNo).also { nomisResponse ->
+      nomisResponse.bookings
+        .find { it.latestBooking }
+        ?.also { latestBooking ->
+          latestBooking.profileDetails
+            .filter { it.type.isPhysicalAttributesProfileType() }
+            .forEach { profileDetail ->
+              profileDetailsSyncService.profileDetailsPhysicalAttributesChanged(profileDetail.type, fromOffenderNo, latestBooking.bookingId, nomisResponse)
+              telemetry += "syncFromOffenderDps_${profileDetail.type}" to "true"
+            }
+        }
+    }
   }
 
-  private suspend fun syncToOffenderDps(toOffenderNo: String, telemetry: MutableMap<String, String>) {
+  private suspend fun syncToOffenderDpsIfChanged(toOffenderNo: String, bookingId: Long, telemetry: MutableMap<String, String>) {
     physicalAttributesNomisApiService.getPhysicalAttributes(toOffenderNo).also { nomisResponse ->
-      val latestBooking = nomisResponse.bookings.find { booking -> booking.latestBooking }
-      if (latestBooking != null &&
-        latestBooking.findLastModifiedPhysicalAttributes().lastModifiedDateTime() > latestBooking.startDateTime
-      ) {
-        physicalAttributesSyncService.physicalAttributesChanged(toOffenderNo, latestBooking.bookingId, nomisResponse)
-        telemetry += "syncToOffenderDps_HEIGHT" to "true"
-        telemetry += "syncToOffenderDps_WEIGHT" to "true"
-      }
+      nomisResponse.bookings
+        .find { it.bookingId == bookingId }
+        ?.takeIf { it.latestBooking }
+        ?.also { latestBooking ->
+          if (latestBooking.findLastModifiedPhysicalAttributes().lastModifiedDateTime() > latestBooking.startDateTime) {
```

Comment:
I'm guessing you have check for items copied by the copy booking NOMIS proc their dates are not greater than the booking start data? E.g no shenanigans where the booking start date is midnight but modified date is greater?

Or is modified date for copied records always null ?

Reply:
Copy booking copies the old audit fields across and booking start time actually includes the time element (booking end doesn't!) - so this if statement *should* always work.